### DISCUSSION TITLE
Is there a way to seperate the query builder of diesel?

### DISCUSSION THREAD

**Discussion by Frodez** [diesel-rs/diesel#2640] [Ideas]
I have found some query builders written in rust, but I think the best one is the query builder of diesel. So if there is a way to seperate it for some purpose, for example, write the dsl and get the raw string of sql, the query builder will be in common use.

**Comment by weiznich** [comment]
I do not think that this is easily possible, as a real query builder is much more than a function that turns ast into a sql string.
For example a query builder has to deal with binding values into prepared statements to prevent sql injections. To do this you need to actually know how to do that for a given connection implementation. Based on this I do not see a way to separate the querybuilder from for example diesels view on how to implement a connection. 
That's one side of the thing, the other is: The query builder is already separated as much as possible from the connection implementations. As long as you implement a bunch of traits (at least `Connection`, possibly `Backend` and everything depended for unsupported database systems) you can use the existing query dsl with any third party connection implementation you like.
